File: Codechef/RESCALC.cpp
#include <algorithm>
#include <cmath>
#include <cctype>
#include <cstdio>
#include <cstring>
#include <iostream>
#include <map>
#include <queue>
#include <set>
#include <string>
#include <vector>

#define FOR(i,a,b) for (int i = a; i <= b; i++)
#define FORN(i,N) for (int i = 0; i < N; i++)
#define FORD(i,a,b) for (int i = a; i >= b; i--)


using namespace std;

typedef long long ll;
typedef unsigned long long ull;
typedef vector<int> vi;
typedef pair<int,int> pii;
typedef vector<pii> vpii;

int main() {
  int T;
  scanf("%d",&T);

  while(T--) {
    int N;
    scanf("%d",&N);
    int maxScore = -1;
    int win = 0;
    bool tie = false;

    FORN(n,N) {
      int C;
      scanf("%d",&C);
      int T[6];
      memset(T,0,sizeof T);
      FORN(i,C) {
        int t;
        scanf("%d",&t);

        T[t-1]++;
      }
      sort(T,T+6);
      /*printf("[");
      FORN(i,6)
        printf("%d ",T[i]);
      printf("]\n");
      */

      int score = C;
      score = max(score,C+T[0]*4);
      score = max(score,C+(T[1]-T[0])*2);
      score = max(score,C+max(0,(T[2]-(T[1]+T[0]))));
      //printf("score = %d\n",score);

      if(maxScore == score) {
        maxScore = score;
        tie = true;
        win = n;
      }
      else if(score > maxScore) {
        tie = false;
        maxScore = score;
        win = n;
      }
    }
    if(tie)
      printf("tie\n");
    else if(win == 0)
      printf("chef\n");
    else
      printf("%d\n",win+1);
  }
  
  return 0;
}

File: Codechef/RESCALC.py


if __name__ == "__main__":
    TC = int(input())

    for t in range(0,TC):
        N = int(input())
        max_score = -1
        tie = False
        win = 0

        for n in range(0,N):
            T = [0] * 6
            C = list(map(int, input().split()))

            for c in range(1,len(C)):
                T[C[c]-1]+=1

            T.sort()

            score = C[0]
            score += T[0]*4
            T[1] -= T[0];
            T[2] -= T[0];
            score += T[1]*2
            T[2] -= T[1];
            score += T[2];

            #print(T)
            #print(score)

            if score > max_score:
                tie = False
                max_score = score
                win = n
            elif score == max_score:
                tie = True
                max_score = score
                win = n

        if tie:
            print("tie")
        elif win == 0:
            print("chef")
        else:
            print(win+1)

